### PR TITLE
SearchShards API added + tests

### PR DESCRIPTION
**Pull request by Zyqsempai**
Added SearchShards API functionality + minimum tests.
https://www.elastic.co/guide/en/elasticsearch/reference/6.0/search-shards.html

### CODE REVIEW

File: search_shards.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
+// Copyright 2012-present Oliver Eilhard. All rights reserved.
+// Use of this source code is governed by a MIT-license.
+// See http://olivere.mit-license.org/license.txt for details.
+
+package elastic
+
+import (
+	"context"
+	"fmt"
+	"github.com/olivere/elastic/uritemplates"
```
Comment (olivere) [inline search_shards.go]: Can you put that into a separate block, just like in the other services, please?

File: search_shards.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+// NewSearchShardsService creates a new SearchShardsService.
+func NewSearchShardsService(client *Client) *SearchShardsService {
+	return &SearchShardsService{
+		client: client,
+	}
+}
+
+// Index sets the names of the indices to restrict the results.
+func (s *SearchShardsService) Index(index ...string) *SearchShardsService {
+	if s.index == nil {
```
Comment (olivere) [inline search_shards.go]: I think the `nil` check is not necessary. It was a misconception I had when I learned Go years ago :-)

**Comment by olivere** [comment]
Hey, thanks for that. Just two minor issues, but I think we can merge it after your fixes.

**Comment by olivere** [comment]
Oh, can you please also add the other parameters (inherited from most services) like `ignore_unavailable` etc. See the [REST API Spec for Search Shards](https://github.com/elastic/elasticsearch/blob/master/rest-api-spec/src/main/resources/rest-api-spec/api/search_shards.json).

**Comment by Zyqsempai** [comment]
Hi Olivere, all done, and also a have in plans to fill up repo with missing APIs in near feature, so if you don't mind, expect few PR on this week.
By the way, thank you for the great library.

**Comment by olivere** [comment]
Sure. Very welcome.

**Pull request merged**